lib/ibge/reader/base.rb

```diff
@@ -35,37 +12,82 @@ def self.read(options = {})
       end
 
       def initialize(options = {})
+        @spreadsheet = {}
+        @rows = {}
+        @selected_sheet = {}
+
+        self.selected_source = self.class.selected_source
+
         @filename = options.fetch(:filename, default_filename)
       end
 
       def read
-        transformed_rows = rows[1..-1].map do |row|
-          transform(row.compact.values_at(*column_range), column_range)
+        transformed_rows = []
+        previous_column_range = nil
+
+        if multiple_sources?
+          loaded_sources = defined_columns.keys.sort
+
+          loaded_sources.each do |loaded_source|
+            self.selected_source = loaded_source
+            @filename = default_filename
+
+            if transformed_rows == []
+              transformed_rows = rows[1..-1].map do |row|
+                transform(row.compact.values_at(*column_range), column_range)
+              end
+
+              transformed_rows = transformed_rows.uniq
+
+              previous_column_range = column_range
+            else
+              transformed_rows.each do |transformed_row|
+                new_row = rows[1..-1].select { |r| r[previous_column_range.first] == transformed_row.values.first }
+                                     .compact
+                                     .uniq { |r| r[0] }
```

Comment:
Place the . on the previous line, together with the method call receiver.<br>Align the operands of an expression in an assignment spanning multiple lines.